tests/hook/enumerate.c

```diff
@@ -0,0 +1,84 @@
+#include "clar_libgit2.h"
+
+static git_repository *g_repo = NULL;
+
+void test_hook_enumerate__initialize(void)
+{
+	g_repo = cl_git_sandbox_init("testrepo");
+}
+
+void test_hook_enumerate__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+	g_repo = NULL;
+}
+
+int test_git_hook_foreach_cb(const char *hook_name, void *payload)
+{
+	git_buf *hook_list = payload;
+	git_buf_puts(hook_list, hook_name);
+	return 0;
+}
+
+void test_hook_enumerate__foreach_hooks(void)
+{
+	git_buf hook_list;
+	git_buf expected_list;
+	git_buf_init(&hook_list, 0);
+	git_buf_init(&expected_list, 0);
```

Comment:
Use `git_buf buf = GIT_BUF_INIT;` instead. You might also consider using `git_vector` instead of `git_buf`, as it is conceptually more correct.